src/matcha_ml/services/analytics_service.py

```diff
@@ -30,10 +30,43 @@ class AnalyticsEvent(str, Enum):
     GET = "get"
 
 
-def execute_analytics_event(
+def _get_state_uuid() -> MatchaResourceProperty:
+    """A function for retrieving the Matcha State UUID.
+
+    Returns:
+        matcha_state_uuid (MatchaResourceProperty): The Matcha State UUID if present.
+
+    Raises:
+    MatchaError where the Matcha state UUID fails validation.
+    """
+    try:
+        matcha_state_service = MatchaStateService()
+    except MatchaError:
+        matcha_state_service = None
```

Comment:
Can we return `None` here and remove checks later?

Reply:
Good idea, implemented.